src/Input/TextField.js

```diff
@@ -181,8 +188,11 @@ class TextField extends React.Component {
       newState.hasValue = nextProps.valueLink.value;
     } else if (hasNewDefaultValue) {
       newState.hasValue = nextProps.defaultValue;
+    } else if (hasNewAutoFocusValue) {
```

Comment:
I think you should not put this in else if because if we have a new value prop and autofocus flag set to true, then we will only be executing the code in hasValueProp if statement only.